Apps/phredmine/redmine_connector.py

```diff
@@ -0,0 +1,996 @@
+
+        ticket_artifact_cef[REDMINE_TICKET_JSON_SUBJECT] = ticket["subject"]
+        ticket_artifact_cef[REDMINE_TICKET_JSON_DESCRIPTION] = ticket["description"]
+
+        ticket_artifact_cef[REDMINE_TICKET_JSON_CREATED_ON] = ticket["created_on"]
+        ticket_artifact_cef[REDMINE_TICKET_JSON_UPDATED_ON] = ticket["updated_on"]
+        ticket_artifact_cef[REDMINE_TICKET_JSON_CLOSED_ON] = ticket.get("closed_on")
+
+        if self._custom_fields_list and ticket.get("custom_fields"):
+            for custom_field in ticket["custom_fields"]:
+                if custom_field["name"] in self._custom_fields_list:
+                    ticket_artifact_cef[custom_field["name"]] = custom_field["value"]
+
+        ticket_artifact["cef"] = ticket_artifact_cef
+
+        artifacts.append(ticket_artifact)
+
+        return artifacts
+
+    def _retrieve_tickets(self, action_result, since, limit=100):
+        """Retrieves tickets from Redmine that recently have been updated"""
+        date = since.strftime("%Y-%m-%dT%H:%M:%S")
+
+        qs = {"updated_on": f">={date}", "limit": limit, "include": "attachments"}
+
+        ret_val, response = self._make_rest_call(
+            "/issues.json", action_result, params=qs
+        )
+
+        if phantom.is_fail(ret_val):
+            message = action_result.get_message()
+            return RetVal(
+                    action_result.set_status(phantom.APP_ERROR,
+                    REDMINE_ERR_RETRIEVE_TICKETS.format(message=message)),
+                    None
+                )
+
+        return ret_val, response["issues"]
+
+    def _retrieve_ticket(self, action_result, id):
+        """Retrieves an individual ticket from Redmine"""
+
+        ret_val, response = self._make_rest_call(
+            f"/issues/{id}.json", action_result, headers=None
+        )
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(
+                phantom.APP_ERROR, REDMINE_ERR_RETRIEVE_TICKET
+            ), None
+        return ret_val, response
+
+    def _retrieve_enumeration_id(self, action_result, enum, enum_key, endpoint):
+        """Given an enum string and a Redmine enumeration, retrieve the corresponding id from the matching endpoint"""
+
+        ret_val, response = self._make_rest_call(
+            endpoint, action_result, params=None, headers=None
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.set_status(
+                phantom.APP_ERROR, REDMINE_ERR_RETRIEVE_DEFINITIONS.format(endpoint=endpoint)
+            )
+
+        try:
+            enum_values = response[enum_key]
+            enum_obj = next(
+                (s for s in enum_values if s["name"].lower() == enum.lower()), None
+            )
+        except Exception as e:
+            self.debug_print(self._get_error_message_from_exception(e))
+            return action_result.set_status(
+                phantom.APP_ERROR, REDMINE_ERR_PROCESSING_RESPONSE
```

Comment:
Please update the error message like an error occurred while processing the response of the Redmine enumeration

Reply:
Modified the error message